README.md

```diff
@@ -1,2 +1,88 @@
-# docker-cuda-tk1
-Docker development environment for the NVIDIA TK1
+# Docker-CUDA-AGX
+Docker development environment for the NVIDIA Jetson AGX
+- Compatible for development for the NVIDIA Jetson Orin NX
+
+Jsoncpp is included in the installation. Please refer to the following instructions for installing the Docker Engine and setting up the Docker CUDA AGX. 
+
+## Installing Docker Engine on Ubuntu
+
+For more information, visit the [dockerdocs](https://docs.docker.com/engine/install/ubuntu/).
+
+Install using the **apt** repository: 
+
+```bash
+# Add Docker's official GPG key:
+sudo apt-get update
+sudo apt-get install ca-certificates curl
+sudo install -m 0755 -d /etc/apt/keyrings
+sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
+sudo chmod a+r /etc/apt/keyrings/docker.asc
+
+# Add the repository to Apt sources:
+echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/ubuntu \
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | \
+  sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+sudo apt-get update
+```
+Install the lastest version:
+
+```bash
+sudo apt-get install docker-ce docker-ce-cli containerd.io docker-buildx-plugin docker-compose-plugin
+```
+
+Ensure the Docker Engine installation is successful by running the `hello-world` image:
+
+```bash
+sudo docker run hello-world
```

Comment:
sudo? I thought the instructions add the current user to the docker group? Doesn't the user need to log out and back in first and then can run without sudo?

Reply:
The first section of instructions instructs the user to use `sudo` to test that the docker engine installed correctly and works properly. 

The second second outlines the post-installation steps which includes adding the current user to the docker group and log out & log back in, so the user does not have to utilize `sudo` anymore. This is tested at the end of the section which instructs the user to run the same "Hello World" line, but without `sudo`.